### PR TITLE
Fix handling of escaped quotes

### PR DESCRIPTION
**Pull request by alexcb**
- this fixes https://github.com/earthly/earthly/issues/859

Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

**Review by vladaionescu — APPROVED** [review]
🆒

File: examples/tests/quotes.earth [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+all:
+    BUILD +test-env-quoting
+    BUILD +test-env-quoting2
+    BUILD +test-multi-line-run-quotes
+    BUILD +test-multi-line-env-quotes
+
+
+test-env-quoting:
+    ENV HI "hello \"world\""
+    RUN test "$HI" = "hello \"world\""
```
[the remark this answers is not in the document]
Comment (alexcb) [reply examples/tests/quotes.earth:13]: good point, I can try hardcoding a sha1sum hash of hello world so we can ensure it's correct?

[the remark this answers is not in the document]
Comment (vladaionescu) [reply examples/tests/quotes.earth:13]: Yeah, good idea

[the remark this answers is not in the document]
Comment (vladaionescu) [reply examples/tests/quotes.earth:13]: Or base64 encoding should work too

File: ast/parser/EarthLexer.g4 [hunk 2]
```diff
@@ -131,7 +131,8 @@ WS_B: WS -> type(WS);
 mode COMMAND_ARGS;
 
 Atom: (RegularAtomPart | QuotedAtomPart)+;
-fragment QuotedAtomPart: ('"' (~'"' | '\\"')* '"');
+fragment QuotedAtomPart: '"' (~('"' | '\\') | '\\' .)* '"';
```
[the remark this answers is not in the document]
Comment (alexcb) [reply ast/parser/EarthLexer.g4:134]: good plan, i'll update it and regenerate.

**Pull request merged**